q2_moshpit/kraken2/bracken.py

```diff
@@ -111,12 +111,34 @@ def _assert_read_lens_available(
         )
 
 
+def _add_unclassified(
+        table: pd.DataFrame,
+        taxonomy: pd.Series,
+        reports: Kraken2ReportDirectoryFormat
+) -> (pd.DataFrame, pd.Series):
+    samples = table.index.tolist()
+    for sample in samples:
+        report_fp = os.path.join(reports.path, f"{sample}.report.txt")
+        with open(report_fp, "r") as f:
+            line = f.readline()
+            if "unclassified" in line:
+                unclassified = int(line.split("\t")[1])
+                if '0' not in taxonomy.index:
+                    taxonomy.loc['0'] = "d__Unclassified"
```

Comment:
This `if` statement can be outside the for-loop, right? It is not dependent to `sample`.

Reply:
It is dependent on samples in a sense that I need to know if there is any sample which has some "unclassified" taxonomy in them. I actually took this opportunity to change the function slightly and decouple parsing the reports from modifying the DataFrame (and incorporating your suggestion) - sorry for making you check again! I hope this code is a bit more transparent about what it's doing though.

Reply:
Thanks for clarifying! The update makes sense.